src/components/FiltersPopover/FiltersPopover.tsx

```diff
@@ -0,0 +1,162 @@
+import clsx from 'clsx';
+import React, { useEffect, useRef } from 'react';
+import styles from './FiltersPopover.module.css';
+import Button from '../Button';
+import ButtonGroup from '../ButtonGroup';
+import Icon from '../Icon';
+import Popover, { PopoverProps } from '../Popover';
+
+export type FiltersPopoverProps = {
+  /**
+   * Form controls, form fields, or other relevant information that will be displayed in the filters popover.
+   */
+  children: React.ReactNode;
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * CSS class names that can be appended to the footer button group.
+   */
+  footerButtonGroupClassName?: string;
+  /**
+   * Indicates status that filters have been selected, influencing toggle button variant.
+   */
+  hasSelectedFilters?: boolean;
+  /**
+   * Callback called when the clear button is called.
+   */
+  onClear?: () => void;
+  /**
+   * Callback called when filters drawer is closed.
+   */
+  onClose?: () => void;
+  /**
+   * Callback called when the apply button is called.
+   */
+  onApply?: () => void;
+  /**
+   * Popover placement options relative to the filters trigger button.
+   */
+  placement?: PopoverProps['placement'];
+  /**
+   * Text to be placed in the button that activates the Filters Popover
+   */
+  triggerText: string;
+};
+
+type FiltersPopoverRenderProps = {
+  /**
+   * Closes popover action render prop passed from Popover parent.
+   */
+  close: (
+    focusableElement?: HTMLElement | React.RefObject<HTMLElement>,
+  ) => void;
+  /**
+   * Popover open status render prop passed from Popover parent.
+   */
+  open: boolean;
+} & Omit<FiltersPopoverProps, 'placement'>;
+
+/**
+ * This helper component passes render props from <Popover> to handle onClose functionality
+ * that HeadlessUI Popover does not provide natively.
+ */
+const FiltersPopoverRender = ({
+  children,
+  className,
+  close,
+  footerButtonGroupClassName,
+  hasSelectedFilters,
+  onApply,
+  onClear,
+  onClose,
+  open,
+  triggerText,
+  ...other
+}: FiltersPopoverRenderProps) => {
+  /**
+   * Hooks to emulate an onClose callback for the Popover.
+   */
+  const firstRender = useRef(true);
+  useEffect(() => {
+    if (firstRender.current) {
+      firstRender.current = false;
+    } else if (!open && onClose) {
+      onClose();
+    }
+  }, [open]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
What's eslint complaining about here?

Reply:
Wants `onClose` as part of the deps, but since `onClose` manages external state, there is a callback loop that occurs. I'll add a comment about this in the code